src/hooks/usePagination.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useLayoutEffect } from 'preact/hooks'
+import { useKeys } from 'hooks'
+
+export const usePagination = (
+  elementRef,
+  step,
+  axis,
+  numOfSection
+) => {
+  const [currentSection, setCurrentSection] = useState(0)
+  const [isLastPage, setIsLastPage] = useState(0)
+  const prop = axis === 'x' ? 'scrollLeft' : 'scrollTop'
+
+  useKeys({
+    ArrowDown: () => {
+      const previous = elementRef.current[prop]
+      elementRef.current[prop] += step
+      const after = elementRef.current[prop]
+
+      // show the next section of the article
+      if (previous === after) {
```

Comment:
On the device, I can skip part of sections, this is the logic responsible.

Reply:
It was actually the CSS that was responsible. I reordered the DIVs and the classes and it fixed it for me. Please test.

Reply:
Confirmed, with this change, I can see it works for my firefox browser, unlike before it works only for first and second page when I paginate.

Can you explain your change?

Reply:
The `page` class needs to be at the top. The name is misleading. It is not related to pagination. It predates it. It represents a page in the general sense, like the search page or the article page.

Article pagination needs to DIVs to work, the outer DIV with the `pages-container` class that handles overflow, and the inner DIV with the 'pages' class that has the column layout. The page class was interfering with the inner one, forcing it to actually be 480px wide. That value is there to trick Firefox, it's not meant to be the actual width.